src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -52,6 +52,14 @@ public abstract class CodeAction
         /// </remarks>
         public virtual string? EquivalenceKey => null;
 
+        /// <summary>
+        /// Identifies the Code Action Provider which produced this Code Action.
+        /// </summary>
+        /// <remarks>
+        /// e.g. 'Add Await Code Action Provider'
+        /// </remarks>
+        public virtual string? ProviderName => null;
```

Comment:
Needs to be added to published API.

Reply:
do we need to actually make this public, or can it be internal (do you need access to this API directly, or is it fine if it's just contained in the code action json payload?

Reply:
I think `internal` should be sufficient. Will double check :)